### PR TITLE
feat: add attack command and event

### PR DESCRIPTION
**Pull request by skbolton**
This command sets up users being able to do initial attacks as well as additional attacks for the round.

Closes: PQ-58

### CODE REVIEW

**Comment by skbolton** [comment]
This event and command made me realize that it is time to introduce policies so that we start enforcing actions better. For example I should only be able to attack for myself. We could start threading the actor through the commands and then include policies in the quest command handlers.

Also very much time to write tests. I figured for now I would get this in so that some UI work could happen around this. Tomorrow I will probably start to write some tests for the quest state.

File: lib/infra/quests/db.ex [hunk 1]
```diff
@@ -23,7 +23,7 @@ defmodule Infra.Quests.Db do
     {:ok, new_quest}
   end
 
-  def write(quest, %Event.AdventurerJoinedParty{} = event) do
+  def write(quest, event) do
```
Comment (skbolton) [inline lib/infra/quests/db.ex:26]: Other than the create event that needs to start up the db server I don't think the events need to be handled differently from the infra perspective. They will always project the change and then update the cache.

File: lib/point_quest/quests/events/adventurer_attacked.ex [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  alias PointQuest.Quests.AttackValue
+
+  @primary_key false
+  embedded_schema do
+    field :quest_id, :string
+    field :adventurer_id, :string
+    field :attack, AttackValue
+  end
+
+  def new!(params) do
```
Comment (skbolton) [inline lib/point_quest/quests/events/adventurer_attacked.ex:14]: Time for a macro to declare these value objects....

File: lib/point_quest/quests/quest.ex [hunk 3]
```diff
@@ -95,6 +95,18 @@ defmodule PointQuest.Quests.Quest do
     }
   end
 
+  def project(%Event.AdventurerAttacked{} = command, %__MODULE__{attacks: attacks} = quest) do
+    # adventurer could be updating their previous attack
```
Comment (skbolton) [inline lib/point_quest/quests/quest.ex:99]: Will test tomorrow

**Comment by JasonNoonan** [comment]
Very much agree on the tests. I was actually thinking I might make that my focus for today's session while trying to fall asleep last night haha

**Review by JasonNoonan — APPROVED** [review]

**Pull request merged**